app/Platform/Actions/ResumePlayerSessionAction.php

```diff
@@ -118,8 +122,36 @@ public function execute(
 
         $user->playerSessions()->save($playerSession);
 
+        // Regenerate the recent players cache so it's primed on the next game page load.
+        $this->regenerateRecentPlayersCache($game->id);
+
         PlayerSessionStarted::dispatch($user, $game, $presence);
 
         return $playerSession;
     }
+
+    /**
+     * Invalidate and pre-warm the recent players cache for a game.
+     *
+     * When a player starts/pings for a game session, we:
+     * 1. Remove the existing cached data.
+     * 2. Pre-warm the cache with fresh data that includes the current player.
+     *
+     * This ensures the game page shows current data without causing a
+     * cache miss for the next visitor.
+     */
+    private function regenerateRecentPlayersCache(int $gameId): void
+    {
+        $cacheKey = "game-recent-players:{$gameId}:10";
```

Comment:
Consider putting this in `CacheKey` to ensure the same key is used in both places.

It would also invert the key to `game:{$gameId}:recent-players:10`.

Reply:
Done in latest.